libcxx/test/std/utilities/meta/meta.unary/meta.unary.prop/reference_constructs_from_temporary.pass.cpp

```diff
@@ -0,0 +1,121 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: std-at-least-c++23
+
```

Comment:
I _think_ we should require Clang 18 here. GCC is still unsupported.

Reply:
> If this is supposed to land before LLVM20 is released we should support Clang 18 but after that we don't. Right?

Hmm... I'm not sure whether some versions of Apple Clang should be considered unsupported.

> What do you mean "GCC is still unsupported"? It looks like GCC14 implemented [P2255R2](https://wg21.link/P2255R2), although I see a compilation error in our tests.

Sorry for misleading... It seems that GCC 14 has supported these 2 intrinsics, but access control was incorrectly ignored. So perhaps we should just skip the `ConvertsToRefPrivate` cases for GCC now.